tsdb/head.go

```diff
@@ -2040,106 +2040,118 @@ func (h *Head) updateWALReplayStatusRead(current int) {
 
 func GenerateTestHistograms(n int) (r []*histogram.Histogram) {
 	for i := 0; i < n; i++ {
-		h := histogram.Histogram{
-			Count:         10 + uint64(i*8),
-			ZeroCount:     2 + uint64(i),
-			ZeroThreshold: 0.001,
-			Sum:           18.4 * float64(i+1),
-			Schema:        1,
-			PositiveSpans: []histogram.Span{
-				{Offset: 0, Length: 2},
-				{Offset: 1, Length: 2},
-			},
-			PositiveBuckets: []int64{int64(i + 1), 1, -1, 0},
-			NegativeSpans: []histogram.Span{
-				{Offset: 0, Length: 2},
-				{Offset: 1, Length: 2},
-			},
-			NegativeBuckets: []int64{int64(i + 1), 1, -1, 0},
-		}
-		if i > 0 {
-			h.CounterResetHint = histogram.NotCounterReset
-		}
-		r = append(r, &h)
+		r = append(r, GenerateTestHistogram(i))
 	}
 	return r
 }
 
+func GenerateTestHistogram(i int) *histogram.Histogram {
+	h := histogram.Histogram{
+		Count:         10 + uint64(i*8),
+		ZeroCount:     2 + uint64(i),
+		ZeroThreshold: 0.001,
+		Sum:           18.4 * float64(i+1),
+		Schema:        1,
+		PositiveSpans: []histogram.Span{
+			{Offset: 0, Length: 2},
+			{Offset: 1, Length: 2},
+		},
+		PositiveBuckets: []int64{int64(i + 1), 1, -1, 0},
+		NegativeSpans: []histogram.Span{
+			{Offset: 0, Length: 2},
+			{Offset: 1, Length: 2},
+		},
+		NegativeBuckets: []int64{int64(i + 1), 1, -1, 0},
+	}
+	if i > 0 {
+		h.CounterResetHint = histogram.NotCounterReset
+	}
+	return &h
+}
+
 func GenerateTestGaugeHistograms(n int) (r []*histogram.Histogram) {
 	for x := 0; x < n; x++ {
-		i := rand.Intn(n)
-		r = append(r, &histogram.Histogram{
-			CounterResetHint: histogram.GaugeType,
-			Count:            10 + uint64(i*8),
-			ZeroCount:        2 + uint64(i),
-			ZeroThreshold:    0.001,
-			Sum:              18.4 * float64(i+1),
-			Schema:           1,
-			PositiveSpans: []histogram.Span{
-				{Offset: 0, Length: 2},
-				{Offset: 1, Length: 2},
-			},
-			PositiveBuckets: []int64{int64(i + 1), 1, -1, 0},
-			NegativeSpans: []histogram.Span{
-				{Offset: 0, Length: 2},
-				{Offset: 1, Length: 2},
-			},
-			NegativeBuckets: []int64{int64(i + 1), 1, -1, 0},
-		})
+		r = append(r, GenerateTestGaugeHistogram(rand.Intn(n)))
 	}
 	return r
 }
 
+func GenerateTestGaugeHistogram(i int) *histogram.Histogram {
+	return &histogram.Histogram{
```

Comment:
Now that we have `GenerateTestHistogram`, we can use that here and set the CounterResetHint before returning. Both are having the same histogram.

Reply:
Yes, I've thought of that, but wasn't sure , working on it.